torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1977,7 +1995,8 @@ void XLATensor::copy_(XLATensor& input, XLATensor& src) {
     if (input.dtype() == src.dtype()) {
       copy_value = src.GetIrValue();
     } else {
-      copy_value = ir::MakeNode<ir::ops::Cast>(src.GetIrValue(), input.dtype());
+      copy_value = ir::MakeNode<ir::ops::Cast>(src.GetIrValue(), input.dtype(),
+                                               src.dtype());
```

Comment:
I understand that passing the `src.dtype()` will truncate the src value based on the src dtype when doing the conversation. However I am wondering why you only chose to pass the `src.dtype()` here?  There are other ops like `ge_`, `eq_` `le_` also calling `ir::MakeNode<ir::ops::Cast>`. Based on the example you gave, shouldn't comparison ops also doing the src dtype check?

Reply:
The ones you mention are the results of the `ComparisonOp()` which is `PRED`.
A cast from `PRED` up, is always OK.
It is needed in `copy_()` because we may be copying a `Byte` tensor into an `Int` one, and the former might have overflown data (onto upper bits).
With the changes to the `CreateFrom()` APIs this *might* not be required anymore, but IMHO it does not harm to be left there.